### PR TITLE
Test/Companies

### PR DESCRIPTION
**Pull request by Aeryle**
- Init prisma on road GetAll, GetOne, post
- remove consoleLog
- PrismaRoads/ProjectsDone
- Create Prisma Client Folder
- connect prisma client
- fix put method bug in users route
- Start Maj test
- Fix review required
- working on prisma routes
- Modify by prisma road
- update test
- Made Try Catch for the post Road
- add returns for the if
- fix
- Routes prisma Companies done
- Working on Prisma Records
- Delete the last log
- Work on tests  - errors to be checked by Aeryle
- 👌 IMPROVE: Improve tests
- 🤖 TEST: Use records endpoints

### CODE REVIEW

File: src/api/companies.js [hunk 1]
```diff
@@ -1,73 +1,84 @@
...
-  const index = companies.findIndex((company) => company.id === +req.params.id);
-  const errorMessage = { message: "Company not found." };
-
-  if (index === -1) {
-    return res.status(404).json(errorMessage);
+router.post("/", async (req, res) => {
+  const { name, logoUrl } = req.body;
+  try {
+    const company = await prisma.company.create({
+      data: {
```
Comment (MatthiasWanner) [inline src/api/companies.js:38]: Does company can be created without name?

File: src/api/companies.js [hunk 2]
```diff
@@ -1,73 +1,84 @@
...
-router.delete("/:id", (req, res) => {
-  const index = companies.findIndex((company) => company.id === +req.params.id);
-  const errorMessage = { message: "Company not found." };
+router.put("/:id", async (req, res) => {
+  const { id } = req.params;
+  const { name, logoUrl } = req.body;
 
-  if (index === -1) {
-    return res.status(404).json(errorMessage);
+  if (!name || !logoUrl) {
```
Comment (MatthiasWanner) [inline src/api/companies.js:53]: Not useful in put route. If undefined, prisma don't update field. This checks have to be in POST logic (logoUrl is nullable).

File: src/api/records.js [hunk 3]
```diff
@@ -1,55 +1,71 @@
...
+    res.status(404).json(error);
+  }
 });
 
-router.post("/", (req, res) => {
-  const { user_id, project_id, step_id, time_slot } = req.body;
-  if (!time_slot) {
-    return res.status(401).json({ message: "Missing fields" });
+router.post("/", async (req, res) => {
+  const { date, timeslot, comment, userId, projectId } = req.body;
```
Comment (MatthiasWanner) [inline src/api/records.js:26]: Please check if all required fields are in the body. Or use Joi in a middleware

File: src/index.js [hunk 4]
```diff
@@ -1,7 +1,8 @@
 const app = require("./app");
 
 const port = process.env.PORT || 5000;
-app.listen(port, () => {
+
+const server = app.listen(port, () => {
```
Comment (MatthiasWanner) [inline src/index.js:5]: Why this change?

File: test/companies.test.js [hunk 5]
```diff
@@ -1,17 +1,19 @@
...
-    "id",
-    "name",
-    "zip_code",
-    "city",
-    "created_at",
-    "updated_at",
-  ];
+const companiesProperties = ["id", "name", "logoUrl", "createdAt", "updatedAt"];
+
+const randomCompany = async (property) => {
```
Comment (MatthiasWanner) [inline test/companies.test.js:7]: Nice! 👍🏻

**Review by MatthiasWanner — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 4 above)
Comment (Aeryle) [reply src/index.js:5]: This was an attempt to fix tests. I didn't rollback it correctly.

**Review by MatthiasWanner — APPROVED** [review]

**Review by Chabelle78 — APPROVED** [review]
